### ISSUE TITLE
Trailing white space is not detected if it's at the end of file

### ISSUE THREAD

**Issue by thakee-kochasoft** [PyCQA/pylint#8076]
### Bug description

If there is a trailing white space at the end of file, pylint isn't detecting it. Run pylint against the bellow code, note that the bellow code snippet contain a space at the last line.

```python
""" Docstring """

def main():
    """Docstring"""
    return 0 
```

### Command used

pylint test.py

### Pylint output

```shell
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### Expected behavior

It should show an error message regarding the trailing white space at the EOF.

### Pylint version

```shell
pylint 2.15.10
astroid 2.13.2
Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]
```

### OS / Environment

OS: Ubuntu 22.04.1 LTS x86_64

**Comment by Pierre-Sassoulas** [comment]
Can you try with ``pylint test.py --enable=all --enable-all-extentions`` ? I suppose the message is deactivated in your env somehow.

**Comment by thakee-kochasoft** [comment]
@Pierre-Sassoulas I've also tried with `--verbose` no error so far...

![Screenshot from 2023-01-19 01-04-26](https://user-images.githubusercontent.com/113889842/213277275-1b6a5ba2-32c6-4b79-9fe4-903426a6d41c.png)


![Screenshot from 2023-01-19 01-02-12](https://user-images.githubusercontent.com/113889842/213276804-7d0f839d-22a8-46cb-b1ef-318158fbb14c.png)

**Comment by Pierre-Sassoulas** [comment]
I can't reproduce either with v2.15.10 or on the main branch even without configuration
```
pylint a.py --enable=all --enable-all-extentions --verbose

No config file found, using default configuration
************* Module a
a.py:5:12: C0303: Trailing whitespace (trailing-whitespace)
a.py:6:0: C0303: Trailing whitespace (trailing-whitespace)

------------------------------------------------------------------

Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)



(venv)  ✘ pierre@pylint  ~/pylint/testwh  ➦ ea12a608b  cat a.py 
""" Docstring """


def main():
    """Docstring"""
    return 0 
 
 ```
I'm also on ubuntu 22.04 so I don't have any leads...

**Comment by clavedeluna** [comment]
I may have reproduced this but just to make sure we're all doing the same

```
""" Docstring """


def main():
    """Docstring"""
    return 0#no extra space here
  # here there is an extra space on the final newline
 ```
![Screen Shot 2023-01-19 at 8 59 44 AM](https://user-images.githubusercontent.com/112832187/213437565-ae86b13b-63e6-4ae8-baba-b1f11e7ca0bf.png)

if this is correct, then I am able to reproduce

```
 pylint test.py --enable=all --enable-all-extensions

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)


 pylint --version
pylint 2.16.0b0
astroid 2.13.2
Python 3.8.9 (default, Apr 13 2022, 08:48:07)
[Clang 13.1.6 (clang-1316.0.21.2.5)]

```

**Comment by Pierre-Sassoulas** [comment]
I'm editing with nano. With a white space only on the last line instead of the two last line I got the expected result too:

```
pylint a.py --enable=all --enable-all-extentions --verbose
No config file found, using default configuration
************* Module a
a.py:6:0: C0303: Trailing whitespace (trailing-whitespace)
------------------------------------------------------------------
Your code has been rated at 5.00/10 (previous run: 0.00/10, +5.00)
```
There's a pre-commit hook to automatically fix trailing-whitespaces with:
```yaml
repos:
  - repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v4.3.0
    hooks:
      - id: trailing-whitespace
```
So I'd suggest using it (or an editor that does it on save as you said) and I don't think I'm going to invest more time to reproduce this.
